servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/OptionalSslNegotiator.java

```diff
@@ -0,0 +1,125 @@
+                                                    executionContext, roConfig,
+                                                    new TcpServerChannelInitializer(roTcpConfig, connectionObserver,
+                                                            executionContext), service,
+                                                    drainRequestPayloadBody, connectionObserver);
+                                        } else {
+                                            return NettyHttpServer.initChannel(channel, executionContext, roConfig,
+                                                    new TcpServerChannelInitializer(roTcpConfig, connectionObserver,
+                                                            executionContext), service,
+                                                    drainRequestPayloadBody, connectionObserver);
+                                        }
+                                    }
+
+                                    // TODO: need to duplicate all but SSL config
+                                    final HttpServerConfig newConfig = new HttpServerConfig();
+                                    if (roConfig.h1Config() != null && roConfig.h2Config() != null) {
+                                        newConfig.httpConfig().protocols(roConfig.h1Config(), roConfig.h2Config());
+                                    } else if (roConfig.h1Config() != null) {
+                                        newConfig.httpConfig().protocols(roConfig.h1Config());
+                                    } else if (roConfig.h2Config() != null) {
+                                        newConfig.httpConfig().protocols(roConfig.h2Config());
+                                    }
+                                    final ReadOnlyHttpServerConfig roNewConfig = newConfig.asReadOnly();
+
+                                    if (roConfig.h2Config() != null) {
+                                        return H2ServerParentConnectionContext.initChannel(listenAddress, channel,
+                                                executionContext, roNewConfig,
+                                                new TcpServerChannelInitializer(roNewConfig.tcpConfig(),
+                                                        connectionObserver, executionContext), service,
+                                                drainRequestPayloadBody, connectionObserver);
+                                    } else {
+                                        return NettyHttpServer.initChannel(channel, executionContext, roNewConfig,
+                                                new TcpServerChannelInitializer(roNewConfig.tcpConfig(),
+                                                        connectionObserver, executionContext), service,
+                                                drainRequestPayloadBody, connectionObserver);
+                                    }
```

Comment:
feel free to ignore: I know it's not often the style in java, but if you put all this in an `else` block it can prevent help bugs since it forces the `if` block to have a return statement on all paths.

Reply:
good idea, thanks!